### PR TITLE
Add whoami API

### PR DESCRIPTION
**Pull request by hexylena**
Simple API route to expose who the current user is.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/configuration.py [hunk 1]
```diff
@@ -20,6 +21,20 @@ def __init__( self, app ):
         super( ConfigurationController, self ).__init__( app )
         self.config_serializer = configuration.ConfigSerializer( app )
         self.admin_config_serializer = configuration.AdminConfigSerializer( app )
+        self.user_manager = users.UserManager( app )
+
+    @expose_api_anonymous
+    def whoami( self, trans, **kwd ):
+        """
+        def whoami( self, trans, **kwd )
+        * GET /api/authenticate/whoami
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/configuration.py]: Should this be just `* GET /api/whoami` ?

File: lib/galaxy/webapps/galaxy/api/configuration.py [hunk 2]
```diff
@@ -20,6 +21,20 @@ def __init__( self, app ):
         super( ConfigurationController, self ).__init__( app )
         self.config_serializer = configuration.ConfigSerializer( app )
         self.admin_config_serializer = configuration.AdminConfigSerializer( app )
+        self.user_manager = users.UserManager( app )
+
+    @expose_api_anonymous
+    def whoami( self, trans, **kwd ):
+        """
+        def whoami( self, trans, **kwd )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/configuration.py]: I don't think this comment line is needed.

File: lib/galaxy/webapps/galaxy/api/configuration.py [hunk 3]
```diff
@@ -20,6 +21,20 @@ def __init__( self, app ):
...
+        * GET /api/authenticate/whoami
+          returns information about the current authenticated user
+
+        :returns: user information in json format
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/configuration.py]: `dictionary with user information` ?

File: lib/galaxy/webapps/galaxy/api/configuration.py (hunk 1 above)
Comment (hexylena) [reply lib/galaxy/webapps/galaxy/api/configuration.py]: Just copied from another set of API docs. Will rewrite, thanks!

File: lib/galaxy/webapps/galaxy/api/configuration.py (hunk 3 above)
Comment (hexylena) [reply lib/galaxy/webapps/galaxy/api/configuration.py]: +1

File: lib/galaxy/webapps/galaxy/api/configuration.py [hunk 4]
```diff
@@ -20,6 +21,19 @@ def __init__( self, app ):
         super( ConfigurationController, self ).__init__( app )
         self.config_serializer = configuration.ConfigSerializer( app )
         self.admin_config_serializer = configuration.AdminConfigSerializer( app )
+        self.user_manager = users.UserManager( app )
+
+    @expose_api_anonymous
+    def whoami( self, trans, **kwd ):
+        """
+        GET /api/authenticate/whoami
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/configuration.py]: Still need to remove `authenticate/` here.

Comment (hexylena) [reply lib/galaxy/webapps/galaxy/api/configuration.py]: Oh! How did I miss that. Cheers.

Comment (hexylena) [reply lib/galaxy/webapps/galaxy/api/configuration.py]: Fixed. I feel a bit like we shouldn't include that and instead have it auto-constructed in the docs?

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/api/configuration.py (hunk 4 above)
Comment (nsoranzo) [reply lib/galaxy/webapps/galaxy/api/configuration.py]: Thanks! That would be good, but not sure how easy it is.

**Pull request merged**